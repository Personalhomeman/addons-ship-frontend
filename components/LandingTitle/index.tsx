import { ReactNode } from 'react';
import css from './style.scss';
import SVG from 'react-svg';

interface LandingTitleProps {
  iconUrl: string;
  children: ReactNode;
}

export default ({ iconUrl, children }: LandingTitleProps) => (
  <div className={css.landingTitle}>
    <SVG src={iconUrl} className={css.icon} />
    <h1 className={css.title}>{children}</h1>
  </div>
);
