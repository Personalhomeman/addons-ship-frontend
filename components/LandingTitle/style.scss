@import '~@/assets/style/dimensions';
@import '~@/assets/style/colors';

.landingTitle {
  display: flex;
  align-items: center;

  .icon {
    margin-right: 2 * $grid;

    svg {
      display: block;
      fill: $white;
    }
  }

  .title {
    min-width: 0;
    word-wrap: break-word;
    color: $white;
    // TODO: replace with element from UI library
    line-height: 1.5;
    font-size: 34px;
    letter-spacing: 0.2px;
    font-weight: 700;
    font-style: normal;
    font-stretch: normal;
  }
}
