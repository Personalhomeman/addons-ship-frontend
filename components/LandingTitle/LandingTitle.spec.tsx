import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import LandingTitle from '.';

describe('LandingTitle', () => {
  it('renders correctly', () => {
    const tree = toJSON(shallow(<LandingTitle iconUrl='https://www.bitrise.io/assets/svg/logo-bitrise.svg'>My app</LandingTitle>));
    expect(tree).toMatchSnapshot();
  });
});
