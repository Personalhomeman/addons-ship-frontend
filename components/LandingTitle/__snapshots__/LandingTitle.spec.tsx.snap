// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingTitle renders correctly 1`] = `
<div
  className="landingTitle"
>
  <ReactSVG
    afterInjection={[Function]}
    beforeInjection={[Function]}
    className="icon"
    evalScripts="never"
    fallback={null}
    loading={null}
    renumerateIRIElements={true}
    src="https://www.bitrise.io/assets/svg/logo-bitrise.svg"
    wrapper="div"
  />
  <h1
    className="title"
  >
    My app
  </h1>
</div>
`;
